Resources/config/services.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <defaults public="false" />
+
+        <service id="joli.command.gif_optimizer" class="Joli\GifExceptionBundle\Command\GifOptimizerCommand">
+            <tag name="console.command" command="jolicode:gifexception:optimize" />
+        </service>
+
+        <service id="gif_exception.listener.replace_image" class="Joli\GifExceptionBundle\EventListener\ReplaceImageListener">
+            <argument type="collection" /> <!-- gif paths -->
+            <argument>%kernel.error_controller%</argument>
+            <argument type="service" id="assets.packages" on-invalid="null" />
+            <tag name="kernel.event_listener" event="kernel.response" method="onKernelResponse" priority="-1000" />
```

Comment:
Something is wrong here. 
1. the class is a subscriber
2. you register it as a listener.

You should use the following 
```xml
<tag name="kernel.event_subscriber" />
```

Reply:
Indeed 😅 
I converted everything into a subscriber